### ISSUE TITLE
WinUI 3 TreeView IsSelected is not persistent when collapsing/expanding tree

### ISSUE THREAD

**Issue by rcohn** [microsoft/microsoft-ui-xaml#8054]
### Describe the bug

Hello -

I am building a WinUI 3 app using TreeView. One of the simplest things that appears not to be working correctly is that a selected TreeViewItem does not stay selected when collapsing the node containing that item and then re-expanding that node.

For example, I have a simple TreeView with a node containing three children items.  After starting the app, I select any of the children items and then collapse the node using the little expand/collapse chevron.  When I re-expand the node, the item is no longer selected (at least it is no longer visibly selected).  The TreeView does not contain any bound data.

In the above description, all selection is through UI interaction with a mouse.

So, is this a bug or do I have to do something further (such as camp on the TreeView.Expanding event and set the item's IsSelected property)?  BTW - I see the same behavior in the TreeView samples in the WinUI 3 Gallery.

Thanks very much for your help.

Best regards,
Robert Cohn

PS - Please let me know if you would need sample code.

### Steps to reproduce the bug

As per the app description above:

- Click on TreeView node to expand it
- Click on one of the child items contained in the node so that it is selected (showing the little selection bar to the left of the item)
- Click on TreeView node to collapse it
- Click on TreeView node to expand it again
- The child item I clicked on is no longer selected (none of the items are selected for that matter)

### Expected behavior

I would expect the clicked child item to still be selected.

### NuGet package version

None

### Windows version

Windows 11 (22H2): Build 22621

### Additional context

I'm using WindowsAppSDK 1.1.4.

**Comment by rcohn** [comment]
BTW - I tried updating to WindowsAppSDK 1.2.221209.1, but that version would not install.  The best I could was update to 1.1.5.

**Comment by ruihuasun** [comment]
I don't think it is a bug but only a service logic, u can try some windows11 native app.

**Comment by rcohn** [comment]
Hello @ruihuasun -

I'm not sure I follow what you mean by "service logic" or what a "windows11 native app" has to do with it.

In a WPF app, I have seen the selected item persist through toggling its parent node.  So I suspect that the WinUI 3 implementation has this bug.  There is nothing in the Microsoft.UI.Xaml.Controls.TreeView class description that indicates selection is not persistent.  As I indicated in my previous comment, I have found a workaround (maybe this is what you are referring to by "service logic") but it seems to be too many hoops to jump through for what should be a supported behavior.

Thank you for your reply.

Regards,
Robert Cohn

**Comment by ruihuasun** [comment]
@rcohn Perhaps the word "Business logic" is more familiar?  What I want to say is _cleaning the selection when collapsing_ is deliberately designed to do so. "native app in windows 11" means the app developed by Microsoft and embedded in windows 11.

**Comment by rcohn** [comment]
Hi @ruihuasun - Thanks for the clarification.

You do have a point.  Depending on where the TreeView is used, the desired behavior may be to clear the selection if the selection is the activation of some function (eg: starting a program from a list of programs).  However, if the TreeView is used in an app where the selection is meant to be an application state (eg: bring up a view indicating a student record) it may be desirable to maintain that state indication until some other state is chosen.

If I were porting a WPF app to WinUI 3, I would expect the WinUI 3 app to match the older app's behavior and not have a new behavior paradigm forced on me.   But that's just my opinion (for what it's worth).

Again, thanks very much for your replies.

Regards,
Robert Cohn
